src/sql/src/pure.rs

```diff
@@ -407,6 +407,16 @@ pub async fn purify_create_source(
                 });
             }
 
+            // Ensure that we have select permissions on all tables; we have to do this before we
+            // start snapshotting because if we discover we cannot `COPY` from a table while
+            // snapshotting, we break the entire source.
```

Comment:
You probably need USAGE privileges on the schema that the table exists in too. Is it possible that this is ever a view and not a table? In that case you need to make sure that the view owner has enough privileges to execute the query within the view.

I don't know the full context of the issue, but you may also need to check that the user has connect privileges on the database that the table lives in.

Reply:
Ah, great--thanks for catching that. Will add it to the tests.

All of these objects we ingest through publications must be tables.